nautobot/extras/factory.py

```diff
@@ -5,12 +5,42 @@
 import faker
 
 from nautobot.core.factory import OrganizationalModelFactory
-from nautobot.extras.models import Status, Tag
-from nautobot.extras.utils import FeatureQuery, TaggableClassesQuery
+from nautobot.extras.models import Role, Status, Tag
+from nautobot.extras.utils import FeatureQuery, RoleModelsQuery, TaggableClassesQuery
 from nautobot.utilities.choices import ColorChoices
 from nautobot.utilities.factory import get_random_instances
 
 
+class RoleFactory(OrganizationalModelFactory):
+    """Role model factory."""
+
+    class Meta:
+        model = Role
+        exclude = (
+            "has_description",
+            "has_weight",
+        )
+
+    name = factory.LazyFunction(
+        lambda: "".join(word.title() for word in faker.Faker().words(nb=2, part_of_speech="adjective", unique=True))
+    )
+    slug = factory.LazyAttribute(lambda role: slugify(role.name))
+    color = factory.Iterator(ColorChoices.CHOICES, getter=lambda choice: choice[0])
+    has_weight = factory.Faker("pybool")
+    weight = factory.Maybe("has_weight", factory.Faker("pyint"), None)
```

Comment:
Interesting that `pyint` by default only generates numbers between 0 and 9999. I was worried that it would potentially generate negative numbers that would be invalid for `weight` as a `PositiveSmallIntegerField`.